File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/Sprint.java
package formula.bollo.app.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import jakarta.persistence.GenerationType;

@Entity
@Table(name = "sprint")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class Sprint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "race_id")
    private Race race;

    @ManyToOne
    @JoinColumn(name = "driver_id")
    private Driver driver;

    @ManyToOne
    @JoinColumn(name = "position_id")
    private SprintPosition position;

    @ManyToOne
    @JoinColumn(name = "season_id")
    private Season season;
}

File: Formula-Bollo/src/shared/models/sprint.ts
import { Driver } from "./driver";
import { Position } from "./position";
import { Race } from "./race";
import { Season } from "./season";

export class Sprint {
  id: number;
  race: Race | null;
  driver: Driver;
  position: Position | null;
  season: Season | undefined;

  constructor(
    id: number,
    race: Race | null,
    driver: Driver,
    position: Position | null,
    season?: Season,
  ) {
    this.id = id;
    this.race = race;
    this.driver = driver;
    this.position = position;
    this.season = season;
  }
}
